bootstrap-app/src/config/bootstrap_migrate_config.clj

```diff
@@ -0,0 +1,58 @@
+(ns config.bootstrap-migrate-config
+  "Provides the configuration for Drift migrations."
+  (:require
+   [clojure.java.jdbc :as j]
+   [cmr.bootstrap.config :as bootstrap-config]
+   [cmr.common.lifecycle :as lifecycle]
+   [cmr.oracle.config :as oracle-config]
+   [cmr.oracle.connection :as oracle]
+   [drift.builder :as drift-builder])
+  (:import
+   (java.sql SQLException)))
+
+(def bootstrap-db-atom (atom nil))
+
+(defn db
+  "Lazily connects to the database and caches it"
+  []
+  (when-not @bootstrap-db-atom
+    (reset! bootstrap-db-atom (lifecycle/start
+                                (oracle/create-db (bootstrap-config/db-spec "bootstrap-migrations"))
+                                nil)))
+  @bootstrap-db-atom)
+
+(defn- maybe-create-schema-table
+  "Creates the schema table if it doesn't already exist."
+  [args]
+  ;; wrap in a try-catch since there is not easy way to check for the existence of the DB
+  (try
+    (j/db-do-commands (db) "CREATE TABLE CMR_BOOTSTRAP.schema_version (version INTEGER NOT NULL, created_at TIMESTAMP(9) WITH TIME ZONE DEFAULT sysdate NOT NULL)")
+    (catch SQLException e
+      ;; 17081 is the error code we get if the table exists and sometimes we also get
+      ;; error message for Universal Connection Pool already exists in the Universal Connection Pool Manager
+      (when-not (or (= 17081 (.getErrorCode e))
+                    (re-matches #"^Unable to start the Universal Connection Pool.*" (.getMessage e)))
+        (throw e)))))
+
+(defn current-db-version []
+  (int (or (:version (first (j/query (db) ["select version from CMR_BOOTSTRAP.schema_version order by created_at desc"]))) 0)))
```

Comment:
This is hard to read.

Reply:
I agree, but this file was simply copied from migrate_config.clj without editing. I'm going to avoid making any changes to the existing code.